plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/condition/TableHasRows.java

```diff
@@ -10,30 +10,93 @@
  ******************************************************************************/ 
 package org.jboss.reddeer.swt.condition;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.hamcrest.Matcher;
 import org.jboss.reddeer.common.condition.AbstractWaitCondition;
 import org.jboss.reddeer.swt.api.Table;
+import org.jboss.reddeer.swt.api.TableItem;
 
 /**
  * Condition is met when table has rows.
  * 
- * @author Rastislav Wagner
+ * @author Rastislav Wagner, Josef Kopriva
  */
 public class TableHasRows extends AbstractWaitCondition {
 	private final Table table;
-
+	private final Matcher<?>[] matchers;
+	private final Integer column;
+	
 	/**
 	 * Constructs TableHasRows wait condition.
 	 * Condition is met when table has any rows.
 	 * 
 	 * @param table table which should contain rows to the condition be met
 	 */
 	public TableHasRows(Table table) {
+		this(table,null,null);
+	}
+	
+	
+	/**
+	 * Constructs TableHasRow wait condition.
+	 * Condition is met when table has specific row matched with matcher.
+	 * 
+	 * @param table table which should contain row to the condition be met
+	 * @param matchers matchers - e.g. Regexmatcher, custom matcher...
+	 */
+	public TableHasRows(Table table, Matcher<?>... matchers) {
+		this(table,null,matchers);
+	}
+	
+	/**
+	 * Constructs TableHasRow wait condition.
+	 * Condition is met when table has specific row and value in specific column matched with matcher.
+	 * 
+	 * @param table table which should contain row to the condition be met
+	 * @param column column in which should be used matchers
+	 * @param matchers matchers - e.g. Regexmatcher, custom matcher...
+	 */
+	public TableHasRows(Table table, Integer column, Matcher<?>... matchers) {
```

Comment:
This a bit confusing since it is not clear how the matchers are used. Is it enough to meet at least one matcher or all matchers must be met? I suggest to remove the option for specifying more matchers and allow to specify only one matcher. Users can use AndMatchers or OrMatchers (I'm not sure what are the exact names).